isthmus/src/main/java/io/substrait/isthmus/expression/WindowFunctionConverter.java

```diff
@@ -4,172 +4,98 @@
     super(functions, typeFactory);
-    this.aggregateFunctionConverter = aggregateFunctionConverter;
   }
 
   @Override
-  protected WindowFunctionInvocation generateBinding(
-      WrappedAggregateCall call,
+  protected Expression.WindowFunctionInvocation generateBinding(
+      WrappedWindowCall call,
       SimpleExtension.WindowFunctionVariant function,
       List<FunctionArg> arguments,
       Type outputType) {
-    AggregateCall agg = call.getUnderlying();
+    RexOver over = call.over;
+    RexWindow window = over.getWindow();
+
+    var partitionExprs =
+        window.partitionKeys.stream()
+            .map(r -> r.accept(call.rexExpressionConverter))
+            .collect(java.util.stream.Collectors.toList());
 
     List<Expression.SortField> sorts =
-        agg.getCollation() != null
-            ? agg.getCollation().getFieldCollations().stream()
-                .map(r -> SubstraitRelVisitor.toSortField(r, call.inputType))
+        window.orderKeys != null
+            ? window.orderKeys.stream()
+                .map(rfc -> toSortField(rfc, call.rexExpressionConverter))
                 .collect(java.util.stream.Collectors.toList())
             : Collections.emptyList();
     Expression.AggregationInvocation invocation =
-        agg.isDistinct()
+        over.isDistinct()
             ? Expression.AggregationInvocation.DISTINCT
             : Expression.AggregationInvocation.ALL;
+
+    WindowBound lowerBound = toWindowBound(window.getLowerBound(), call.rexExpressionConverter);
+    WindowBound upperBound = toWindowBound(window.getUpperBound(), call.rexExpressionConverter);
+
     return ExpressionCreator.windowFunction(
         function,
         outputType,
         Expression.AggregationPhase.INITIAL_TO_RESULT,
         sorts,
         invocation,
+        partitionExprs,
+        lowerBound,
+        upperBound,
         arguments);
   }
 
-  public Expression.Window convert(
-      RelNode input,
-      Type.Struct inputType,
+  public Optional<Expression.WindowFunctionInvocation> convert(
       RexOver over,
       Function<RexNode, Expression> topLevelConverter,
       RexExpressionConverter rexExpressionConverter) {
-
-    var lowerBound = toWindowBound(over.getWindow().getLowerBound(), rexExpressionConverter);
-    var upperBound = toWindowBound(over.getWindow().getUpperBound(), rexExpressionConverter);
-    var sqlAggFunction = over.getAggOperator();
-    var argList =
-        over.getOperands().stream()
-            .map(r -> ((RexSlot) r).getIndex())
-            .collect(java.util.stream.Collectors.toList());
-    boolean approximate = false;
-    int filterArg = -1;
-    var call =
-        AggregateCall.create(
-            sqlAggFunction,
-            over.isDistinct(),
-            approximate,
-            over.ignoreNulls(),
-            argList,
-            filterArg,
-            null,
-            RelCollations.EMPTY,
-            over.getType(),
-            sqlAggFunction.getName());
```

Comment:
The usage of this API meant that calls like `ntile(4)` were not supported, a `4` would be encoded as a `RexLiteral` which is not a RexSlot. Switching away from this handling enables those queries, and in fact I was able to enable the test for the `ntile` query in `WindowFunctionTest`.